unittests/runtime/CMakeLists.txt

```
@@ -83,6 +83,13 @@ if(("${SWIFT_HOST_VARIANT_SDK}" STREQUAL "${SWIFT_PRIMARY_VARIANT_SDK}") AND
     endif()
   endif()
 
+  # Check that the tests are using the async calling convention
+  # correctly, and that musttail isn't missed by Clang/LLVM.
+  if(NOT SWIFT_COMPILER_IS_MSVC_LIKE)
+    set(CMAKE_CXX_FLAGS
+          "${CMAKE_CXX_FLAGS} -mllvm -enable-swifttailcc-musttail-check")
```

A more idiomatic way to set these flags for `SwiftRuntimeTests` is to use `target_compile_options` and move this `if` before the finale `endif` in the file

```suggestion
    target_compile_options(SwiftRuntimeTests
      PUBLIC "-mllvm;-enable-swifttailcc-musttail-check")
```

This way we would match how the other properties for `SwiftRuntimeTests` are set after its definition, and we would suggest that those flags are not meant to be set globally for all targets.

---

Fixed.